tests/lib/collectors/aws_snapshot_s3_copy.py

```diff
@@ -0,0 +1,304 @@
+  """Mock the boto3 api calls for specified client methods.
+
+  Args:
+    operation_name: The AWS API operation.
+    kwarg: Args to pass to the method.
+  """
+  if operation_name == 'DescribeSnapshots':
+    return FAKE_DESCRIBE_SNAPSHOTS
+  if operation_name == 'DescribeAvailabilityZones':
+    return FAKE_DESCRIBE_AVAILABILITY_ZONES
+  if operation_name == 'DescribeSubnets':
+    return FAKE_DESCRIBE_SUBNETS
+  if operation_name == 'ListBuckets':
+    return FAKE_LIST_BUCKETS_RESPONSE
+  if operation_name == 'CreateBucket':
+    return FAKE_CREATE_BUCKET_RESPONSE
+  return orig(self, operation_name, kwarg)
+
+class AWSSnapshotS3CopyCollectorTest(unittest.TestCase):
+  """Tests for the AWSSnapshotS3CopyCollector."""
+
+  def testInitialization(self):
+    """Tests that the collector can be initialized."""
+    test_state = state.DFTimewolfState(config.Config)
+    collector = aws_snapshot_s3_copy.AWSSnapshotS3CopyCollector(test_state)
+    self.assertIsNotNone(collector)
+
+  def testSetUp(self):
+    """Tests SetUp of the collector."""
+    test_state = state.DFTimewolfState(config.Config)
+    snapshots_expected = ['snap-01234567', 'snap-12345678']
+
+    # Subnet is optional - test with it.
+    collector = aws_snapshot_s3_copy.AWSSnapshotS3CopyCollector(test_state)
+    collector.SetUp(','.join([snapshot['SnapshotId']\
+          for snapshot in FAKE_DESCRIBE_SNAPSHOTS['Snapshots']]),
+        FAKE_BUCKET,
+        FAKE_REGION,
+        FAKE_SUBNET)
+
+    state_snaps = [snap.snap_id for snap in \
+        test_state.GetContainers(containers.AWSSnapshot)]
+
+    self.assertEqual(snapshots_expected,state_snaps)
+    self.assertEqual(FAKE_REGION, collector.region)
+    self.assertEqual(FAKE_BUCKET, collector.bucket)
+    self.assertEqual(FAKE_SUBNET, collector.subnet)
+
+    # Subnet is optional - test without it.
+    test_state = state.DFTimewolfState(config.Config)
+    collector = aws_snapshot_s3_copy.AWSSnapshotS3CopyCollector(test_state)
+    collector.SetUp(','.join([snapshot['SnapshotId']\
+          for snapshot in FAKE_DESCRIBE_SNAPSHOTS['Snapshots']]),
+        FAKE_BUCKET,
+        FAKE_REGION)
+
+    state_snaps = [snap.snap_id for snap in \
+        test_state.GetContainers(containers.AWSSnapshot)]
+
+    self.assertEqual(snapshots_expected, state_snaps)
+    self.assertEqual(FAKE_REGION, collector.region)
+    self.assertEqual(FAKE_BUCKET, collector.bucket)
+    self.assertEqual(None, collector.subnet)
+
+  @mock.patch('boto3.session.Session._setup_loader')
+  def testPickAvailabilityZone(self, mock_loader):
+    """Test the utility funciton that picks an anailability zone in the
+    region to use."""
+    mock_loader.return_value = None
+
+    test_state = state.DFTimewolfState(config.Config)
+    collector = aws_snapshot_s3_copy.AWSSnapshotS3CopyCollector(test_state)
+
+    with mock.patch('botocore.client.BaseClient._make_api_call',
+      new=MockMakeAPICall):
+
+      collector.SetUp(','.join([snapshot['SnapshotId']\
```

Comment:
nit: I would probably split this line (and other similar lines in this file) up for readability.